java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/KinesisProducer.java

```diff
@@ -132,7 +150,12 @@ public void run() {
                     } else if (m.hasMetricsResponse()) {
                         onMetricsResponse(m);
                     } else {
-                        log.error("Unexpected message type from child process");
+                        // clear the future here as well since the native core has exhausted its retries.
+                        getFuture(m);
```

Comment:
Can we check if m.getSourceId is not null here?

Reply:
its a primitive long type hence cant be null